client/core/core.go

```diff
@@ -1052,11 +1052,28 @@ func (c *Core) connectedWallet(assetID uint32) (*xcWallet, error) {
 	return wallet, nil
 }
 
+// connectWallet connects to wallet and validate the known deposit address
+// after successfull connectection, if deposit address does not belong to
+// wallet it generates new address and updates xcWallet and dbWallet, therefore
+// it might hold the wallet lock
 func (c *Core) connectWallet(w *xcWallet) error {
 	err := w.Connect(c.ctx)
 	if err != nil {
 		return codedError(connectWalletErr, err)
 	}
+	// Check if known address belongs to connected wallet
+	mine, err := w.ValidateAddress(w.address)
+	if err != nil {
+		return err
+	}
+	if !mine {
+		c.log.Infof("[%v]: Deposit address %v does not belong to connected wallet"+
+			", generating new address", unbip(w.AssetID), w.address)
```

Comment:
This should be uncommon, so I would not object to `Warnf`.  Also, since you're logging the old deposit address, you should log the new one too.